src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -949,11 +949,24 @@ export function isOccupied(matrix: string[][], position: Position, chartDimensio
     return occupied;
 }
 
-export function calculateLayout(visibleAttributes: ChartMeta[], cols: number): Layout[] {
+export function calculateLayout(visibleAttributes: ChartMeta[], cols: number, currentGridLayout?: Layout[], currentGridElement?: ChartMeta): Layout[] {
     let layout: Layout[] = [];
     let matrix = [new Array(cols).fill('')] as string[][];
+    let sortedVisibleAttributes = visibleAttributes.sort((a, b) => b.priority - a.priority);
 
-    _.forEach(visibleAttributes.sort((a, b) => b.priority - a.priority), (chart: ChartMeta) => {
+    if (currentGridLayout && currentGridLayout.length > 0) {
+        if (currentGridElement){
```

Comment:
I would suggest to separate out the currentGridElement out from sortedVisibleAttributes if available. Would be easier to deal with later on when currentGridElement becomes an array.

Reply:
Then you also avoid to check `currentGridElement` multiple times.